webui/react/src/stores/cluster.tsx

```diff
@@ -93,17 +92,16 @@ const clusterStatusText = (
   )}%`;
 };
 
-const updateIfChanged = <T, V extends WritableObservable<T>>(o: V, next: T) =>
-  o.update((prev) => (_.isEqual(prev, next) ? prev : next));
-
 class ClusterStore extends PollingStore {
-  #agents: WritableObservable<Loadable<Agent[]>> = observable(NotLoaded);
-  #resourcePools: WritableObservable<Loadable<ResourcePool[]>> = observable(NotLoaded);
-  #unboundResourcePools: WritableObservable<Loadable<ResourcePool[]>> = observable(NotLoaded);
-  #resourcePoolBindings: WritableObservable<Map<string, number[]>> = observable(Map());
+  #agents = deepObservable<Loadable<Agent[]>>(NotLoaded);
```

Comment:
With the introduction of `asValueObject` and `immutableObservable` it does open a question of whether we should be using `List` from immutable.js.

Reply:
the top value being observed here isn't an immutable collection, though -- it's a Loadable of an immutable collection. The issue here is that neither observable type works for that combo. `deepObservable` will trip over the immutable value because two collections that have the same values can have different state, and without a way of describing a immutable collection in `io-ts`, there isn't a way of detecting that when putting together the equality function in `asValueObject`.

Reply:
> without a way of describing a immutable collection in io-ts

Worth noting that this would be pretty easy to add. Obviously out of scope for this PR but I think it would increase consistency.